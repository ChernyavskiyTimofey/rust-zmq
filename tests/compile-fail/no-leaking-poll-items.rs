fn main() {
    let context = zmq2::Context::new();
    let _poll_item = {
        let socket = context.socket(zmq2::PAIR).unwrap();
        socket.as_poll_item(zmq2::POLLIN)
    }; //~^ ERROR `socket` does not live long enough [E0597]
}
