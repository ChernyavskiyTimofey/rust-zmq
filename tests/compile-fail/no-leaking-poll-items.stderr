error[E0597]: `socket` does not live long enough
 --> $DIR/no-leaking-poll-items.rs:5:9
  |
3 |     let _poll_item = {
  |         ---------- borrow later stored here
4 |         let socket = context.socket(zmq2::PAIR).unwrap();
5 |         socket.as_poll_item(zmq2::POLLIN)
  |         ^^^^^^ borrowed value does not live long enough
6 |     }; //~^ ERROR `socket` does not live long enough [E0597]
  |     - `socket` dropped here while still borrowed
